streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -1062,6 +1062,25 @@ object ZSink extends ZSinkPlatformSpecificConstructors with Serializable {
 
   implicit class InvariantOps[R, E, A0, A, B](val sink: ZSink[R, E, A0, A, B]) extends AnyVal { self =>
 
+    /**
+     * Drops the first `n`` elements from the sink.
+     */
+    final def drop(n: Long): ZSink[R, E, A0, A, B] =
+      new ZSink[R, E, A0, A, B] {
+        type State = (sink.State, Long)
+
+        // Cast is redundant but required for Scala 2.11
+        val initial = sink.initial.map((_, 0L)).asInstanceOf[ZIO[R, E, this.State]]
+
+        def step(state: State, a: A) =
+          if (state._2 != n) UIO.succeed((state._1, state._2 + 1))
+          else sink.step(state._1, a).map((_, state._2))
+
+        def extract(state: State) = sink.extract(state._1)
+
+        def cont(state: State) = sink.cont(state._1)
```

Comment:
I would think the sink needs to continue for at least `n` iterations and then "switch over" to the underlying sink.

Reply:
or when both continue?
```scala
state._2 < n && sink.cont(state._1)
```